src-tauri/src/lib/utils.rs

```diff
@@ -751,6 +760,114 @@ fn is_retryable_error(error: &io::Error) -> bool {
     }
 }
 
+/// Returns the base domain (scheme + host + optional port) from a full URL.
```

Comment:
either this comment or the function is not correct as this implementation toss the port out